fencing/commands.c

```diff
@@ -987,22 +893,35 @@ schedule_internal_command(const char *origin,
     schedule_stonith_command(cmd, device);
 }
 
-gboolean
-string_in_list(GListPtr list, const char *item)
+/* very inefficient - would need improvement when used more often */
+char *
+list_to_string(GListPtr list, const char *delim, gboolean terminate_with_delim)
 {
     int lpc = 0;
     int max = g_list_length(list);
+    int delim_len = delim?strlen(delim):0;
+    int alloc_size = 1 + (max?((max-1+(terminate_with_delim?1:0))*delim_len):0);
+    char *rv;
 
     for (lpc = 0; lpc < max; lpc++) {
         const char *value = g_list_nth_data(list, lpc);
 
-        if (safe_str_eq(item, value)) {
-            return TRUE;
-        } else {
-            crm_trace("%d: '%s' != '%s'", lpc, item, value);
+        alloc_size += strlen(value);
+    }
+    rv = calloc(1, alloc_size);
```

Comment:
The old question when callocing a string: array of chars or one string ;-)
Just intended to make it consistent within pacemaker and thought I had copied it from somewhere - but might remember it wrongly.

Reply:
Yep, there are plenty of examples to lead someone wrong in pacemaker. :) I've been updating them as I change code around them, but there are still a bunch to do

Reply:
Problem of the having ugly stuff aged out is that the list of things to look at when changing code for other reason is getting quite long and is inconsistently distributed over a lot of people's minds.
Maybe it would be useful to keep a list of that stuff in parallel to the coding-conventions.
In theory that would maybe even be possible (at least for some easy things like formating) to be caught by git on push/pull-request - haven't used it but I think it is there.